### DISCUSSION TITLE
Behaviour upon receiving empty eventData

### DISCUSSION THREAD

**Discussion by 01ste02** [zwave-js/zwave-js#7525] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
When receiving the following event:
```
12/28/2024, 9:31:05 PM - notification
Arg 0:
111
Arg 1:
└─eventType: 3
└─dataType: 0
└─eventData
└─eventTypeLabel: Disarm all
└─dataTypeLabel: None
```

We get an empty payload, even though the dataType is set to 0. Is this allowed? This is then interpreted as 0x00 when stringifying the JSON data, resulting in the following parsing:
```
2024-12-28T19:33:15.859Z DRIVER « [Node 003] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 242
                                    │ security class:  S2_AccessControl
                                    └─[EntryControlCCNotification]
                                        sequence number: 5
                                        data type:       0
                                        event type:      3
                                        event data:      0x00
```

Is this expected behavior on your part, or is it undefined/unexpected?

Anything else we should know?

This is also discussed here:
https://github.com/home-assistant/core/issues/134172

With more in-depth analysis in https://github.com/home-assistant/core/issues/134172#issuecomment-2564640236

### Software versions

Driver (node-zwave-js):  14.3.7

Z-Wave JS UI:  9.27.8

Home Assistant Z-Wave Integration: 1.40.2

### Z-Wave Controller (Stick/Dongle/...)

AEON Labs Z stick Gen5 USB Controller v1.1

### Device information

Manufacturer: Philio Tech
Model name: PSK01
Node ID: 3

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile
[home-assistant_zwave_js_2024-12-28T19-33-19.766Z.log](https://github.com/user-attachments/files/18270795/home-assistant_zwave_js_2024-12-28T19-33-19.766Z.log)

[a comment by an automated account is not shown]

**Reply by 01ste02** [reply]
Updated the logfile to the correct one

**Comment by kpine** [comment]
> We get an empty payload, even though the dataType is set to 0. Is this allowed? This is then interpreted as 0x00 when stringifying the JSON data, resulting in the following parsing

To clarify (as I suggested submitting the issue), there is **not** an empty payload, as that is what one would actually expect with the `NA` data type. The event data contains a single byte set to `0` (e.g. `"\0"`). The question is whether Z-Wave JS should omit the event data when the type is `NA`, as the specs would indicate that is not valid, or if you just want to pass the data as-is.
